lib/smart_answer/money.rb

```diff
@@ -9,15 +9,8 @@ class Money
     attr_reader :value
 
     def initialize(raw_input)
-      if raw_input.is_a?(Numeric)
-        @value = BigDecimal.new(raw_input.to_s)
-      else
-        raw_input = raw_input.to_s.delete(',').gsub(/\s/, '')
-        if !self.class.valid?(raw_input)
-          raise InvalidResponse, "Sorry, I couldn't understand that number. Please try again.", caller
-        end
-        @value = BigDecimal.new(raw_input.to_s)
-      end
+      number = self.class.raw_input_to_number(raw_input)
+      @value = BigDecimal.new(number.to_s)
```

Comment:
@alecgibson indeed without the `.to_s` call it is returning `ArgumentError: can't omit precision for a Float.`. It probably guesses the precision needed if the argument is a String. That's a little bit weird. Anyway I would leave it as it is for moment, if it is ok for you.